### PR TITLE
Add a script to create test themes

### PR DESCRIPTION
**Pull request by Arkham**
### WHY are these changes introduced?

It's currently hard to test all possible configurations of how the shopify tooling is installed for themes.

### WHAT is this pull request doing?

Adds a node script to create test themes using homebrew, local cli repo, and npm global nightly packages. 

### How to test your changes?

First of all unset the following env vars:
- `unset GEM_HOME`
- `unset GEM_PATH`

We need to do this otherwise the shadowenv variables are messing with where gems are looked up. I am investigating if we should always pass it when running `bundle` commands.

Then try once for each supported way:
- local
  - `node bin/create-test-theme.js -s <YOUR_STORE> --cleanup -i local`
  - Open http://127.0.0.1:9292/
  - Change the background color of the first div element in `sections/announcement-bar.liquid` to `color-inverse`
  - Check that the theme is hot reloaded
  - press `Ctrl+C` to continue
  - script should continue and terminate successfully
- homebrew
  - `node bin/create-test-theme.js -s <YOUR_STORE> --cleanup -i homebrew`
  - Open http://127.0.0.1:9292/
  - Change the background color of the first div element in `sections/announcement-bar.liquid` to `color-inverse`
  - Check that the theme is hot reloaded
  - press `Ctrl+C` to continue
  - script should continue and terminate successfully
- npm globally
  - `node bin/create-test-theme.js -s <YOUR_STORE> --cleanup -i npm-global`
  - Open http://127.0.0.1:9292/
  - Change the background color of the first div element in `sections/announcement-bar.liquid` to `color-inverse`
  - Check that the theme is hot reloaded
  - press `Ctrl+C` to continue
  - script should continue and terminate successfully

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli-kit/src/public/node/ruby.ts [hunk 1]
```diff
@@ -441,5 +441,8 @@ async function localBundleInstall(directory: string): Promise<void> {
   await exec(bundle, ['config', 'set', '--local', 'path', directory], {
     cwd: directory,
   })
-  await exec(bundle, ['install', '--without', 'development', 'test'], {cwd: directory})
+  await exec(bundle, ['config', 'set', '--local', 'without', 'development:test'], {
```
Comment (Arkham) [inline packages/cli-kit/src/public/node/ruby.ts:444]: bundler suggested me to set the config in this way :)

**Comment by isaacroldan** [comment]
Found this issue while trying the global installation:
Captura de pantalla 2023-03-08 a las 14 19 25

**Comment by Arkham** [comment]
Did you unset the two GEM_ env vars?

**Comment by isaacroldan** [comment]
I did :/

**Review by isaacroldan — APPROVED** [review]

**Pull request merged**